preprocessing/nextclade/tests/test_nextclade_preprocessing.py

```diff
@@ -0,0 +1,562 @@
+                "NPEbolaSudan": get_ebola_sudan_aa(get_sequence_with_mutation("single"), "NP"),
+                "VP35EbolaSudan": get_ebola_sudan_aa(get_sequence_with_mutation("single"), "VP35"),
+            },
+            aminoAcidInsertions={},
+        ),
+    ),
+    Case(
+        name="with insertion",
+        input_metadata={},
+        input_sequence={"main": get_sequence_with_insertion("single")},
+        accession_id="1",
+        expected_metadata={
+            "completeness": 1.0,
+            "totalInsertedNucs": 3,
+            "totalSnps": 0,
+            "totalDeletedNucs": 0,
+            "length": len(get_sequence_with_insertion("single")),
+        },
+        expected_errors=[],
+        expected_warnings=[],
+        processed_alignment=ProcessedAlignment(
+            unalignedNucleotideSequences={"main": get_sequence_with_insertion("single")},
+            alignedNucleotideSequences={"main": get_consensus_sequence("single")},
+            nucleotideInsertions={"main": ["2671:GAC"]},
+            alignedAminoAcidSequences={
+                "NPEbolaSudan": get_ebola_sudan_aa(get_consensus_sequence("single"), "NP"),
+                "VP35EbolaSudan": get_ebola_sudan_aa(get_consensus_sequence("single"), "VP35"),
+            },
+            aminoAcidInsertions={"NPEbolaSudan": ["738:D"]},
+        ),
+    ),
+    Case(
+        name="with deletion",
+        input_metadata={},
+        input_sequence={"main": get_sequence_with_deletion("single")},
+        accession_id="1",
+        expected_metadata={
+            "completeness": 1.0,
+            "totalInsertedNucs": 0,
+            "totalSnps": 0,
+            "totalDeletedNucs": 3,
+            "length": len(get_consensus_sequence("single")) - 3,
+        },
+        expected_errors=[],
+        expected_warnings=[],
+        processed_alignment=ProcessedAlignment(
+            unalignedNucleotideSequences={"main": get_sequence_with_deletion("single")},
+            alignedNucleotideSequences={"main": get_sequence_with_deletion("single", aligned=True)},
+            nucleotideInsertions={"main": []},
+            alignedAminoAcidSequences={
+                "NPEbolaSudan": get_ebola_sudan_aa(
+                    get_sequence_with_deletion("single", aligned=True), "NP"
+                ),
+                "VP35EbolaSudan": get_ebola_sudan_aa(
+                    get_sequence_with_deletion("single", aligned=True), "VP35"
+                ),
+            },
+            aminoAcidInsertions={},
+        ),
+    ),
+    Case(
+        name="with failed alignment",
+        input_metadata={},
+        input_sequence={"main": get_invalid_sequence()},
+        accession_id="1",
+        expected_metadata={
+            "completeness": None,
+            "totalInsertedNucs": None,
+            "totalSnps": None,
+            "totalDeletedNucs": None,
+            "length": 53,
+        },
+        expected_errors=[
+            ProcessingAnnotationTestCase(
```

Comment:
make an issue: we probably shouldn't show this error for the single segment case

Reply:
https://github.com/loculus-project/loculus/issues/4765